staging/src/k8s.io/apimachinery/pkg/util/taints/taints.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// package taints implements utilites for working with taints
+package taints
+
+import (
+	"strings"
+
+	"k8s.io/api/core/v1"
```

Comment:
this repo is for universal apimachinery pkgs. we dont want any `k8s.io/api` dependency here

Reply:
little knowledge about it. Do you have any suggestion about where to put it?  how about `vendor/k8s.io/util`?

Reply:
> how about vendor/k8s.io/util?

mmmmmmm...idk @sttts any thought?

Reply:
In order to remove `pkg/controller` dependencies from `k8s.io/kubernetes/pkg/cloudprovider/providers`, see https://github.com/kubernetes/kubernetes/pull/70005/files#r226647860, function `AddOrUpdateTaintOnNode()` https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/controller_utils.go#L903 must move out `k/k`, but innner function https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/controller_utils.go#L927 depend on `k8s.io/kubernetes/pkg/util/taints`. So this PR done something about it

Reply:
We don't have a place for helpers like this. What would happen if we move them into k8s.io/cloud-provider? We could still use them from k/k code.

Reply:
`k8s.io/cloud-provider` works for me! It might be better to just duplicate the 1 method we need rather than moving this whole package over there. Once there's a better place for helpers like this we can remove the duplicate code. It's a bit hacky but it'll unblock us for the time being.

Reply:
https://github.com/kubernetes/kubernetes/pull/73491 is going to unblock us for now. Would be great if this PR can refactor things a bit so that taint util doesn't have to be duplicated in k8s.io/cloud-provider :)

Reply:
yes, great work :+1: